### PR TITLE
Features/1.0.8

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

**Comment by alexbourret** [comment]
- Improved logging
- Adds form digest value to posts
- Adds retries on get & post following connection reset by peer, throttling events (429 errors)
- Adds session reset following 403 errors

File: python-lib/robust_session.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        """
+        attempt_number = 0
+        successful_request = False
+        while (not successful_request) and (attempt_number <= self.max_retries):
+            attempt_number += 1
+            if verb == "get":
+                response = self.retry(self.session.get, **kwargs)
+            else:
+                response = self.retry(self.session.post, **kwargs)
+            if response.status_code == 403:
```
Comment (andreybavt) [inline python-lib/robust_session.py:70]: Out of curiosity, why does Microsoft reject our calls? Do they have a policy on how often their API can be called?

**Review by andreybavt — CHANGES_REQUESTED** [review]
Generally looks good, I added a few comments.
However I'm a newbie because I never used Sharepoint.

File: python-lib/sharepoint_client.py [hunk 2]
```diff
@@ -583,12 +702,50 @@ def post(self, url, headers=None, json=None, data=None):
...
         return "Bearer {}".format(self.sharepoint_access_token)
 
+    def get_form_digest_value(self):
+        logger.info("Getting form digest value")
+        session = RobustSession(session=requests, status_codes_to_retry=[429])
+        session.update_settings(
+            max_retries=SharePointConstants.MAX_RETRIES,
+            base_retry_timer_sec=SharePointConstants.WAIT_TIME_BEFORE_RETRY_SEC
+        )
+        headers = {
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/sharepoint_client.py:724]: 👍

File: python-connectors/sharepoint-online_lists/connector.py [hunk 3]
```diff
@@ -42,16 +46,29 @@ def get_read_schema(self):
         dss_columns = []
         self.column_ids = {}
         self.column_names = {}
+        self.column_entity_property_name = {}
         for column in sharepoint_columns:
+            logger.info("get_read_schema:{}/{}/{}/{}/{}/{}".format(
+                column[SharePointConstants.TITLE_COLUMN],
```
[the remark this answers is not in the document]
Comment (andreybavt) [reply python-connectors/sharepoint-online_lists/connector.py:52]: ok, it was a cosmetic preference

**Review by andreybavt — APPROVED** [review]

**Pull request merged**